File: 단계별로 풀어보기/백트래킹/15652/N과 M (4).py
import sys

def func(cnt, start):
    if cnt == M:
        for i in range(M):
            sys.stdout.write('%d ' %arr[i])
        sys.stdout.write('\n')
    else:
        for i in range(start, N):
            arr[cnt] = i+1
            func(cnt+1, i)

N, M = map(int,sys.stdin.readline().split())
arr = [0 for i in range(8)]
func(0, 0)

File: 단계별로 풀어보기/백트래킹/15652/N과 M (4).cpp
#include <iostream>
#define MAX 7 
using namespace std;

int arr[MAX] = {0,};
int N, M;

int func(int cnt, int start){
    
   if(cnt == M){
       for(int i = 0; i < M; i++){
           cout << arr[i] << ' ';
       }
       cout << '\n';
   } else{
        for(int i = start; i < N; i++){
            arr[cnt] = i+1;
            func(cnt+1, i);
        } 
   }
    
    
    return 0;
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    cin >> N >> M;    
    func(0, 0);
    
    return 0;
}
